subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPlugin.java

```diff
@@ -381,28 +352,23 @@ private void registerSoftwareComponents(Project project) {
         project.getComponents().add(java);
     }
 
-    private void addJar(Configuration configuration, PublishArtifact jarArtifact) {
+    private static void addJar(Configuration configuration, PublishArtifact jarArtifact) {
         ConfigurationPublications publications = configuration.getOutgoing();
 
         // Configure an implicit variant
         publications.getArtifacts().add(jarArtifact);
         publications.getAttributes().attribute(ArtifactTypeDefinition.ARTIFACT_TYPE_ATTRIBUTE, ArtifactTypeDefinition.JAR_TYPE);
     }
 
-    private void addRuntimeVariants(Configuration configuration, PublishArtifact jarArtifact, final SourceSet sourceSet, final Provider<ProcessResources> processResources) {
-        ConfigurationPublications publications = configuration.getOutgoing();
-
-        // Configure an implicit variant
-        publications.getArtifacts().add(jarArtifact);
-        publications.getAttributes().attribute(ArtifactTypeDefinition.ARTIFACT_TYPE_ATTRIBUTE, ArtifactTypeDefinition.JAR_TYPE);
-
+    private void addRuntimeVariants(Project project, Configuration configuration, final SourceSet sourceSet) {
         // Define some additional variants
         jvmServices.configureClassesDirectoryVariant(sourceSet.getRuntimeElementsConfigurationName(), sourceSet);
-        NamedDomainObjectContainer<ConfigurationVariant> runtimeVariants = publications.getVariants();
+        NamedDomainObjectContainer<ConfigurationVariant> runtimeVariants = configuration.getOutgoing().getVariants();
```

Comment:
Minor point: I think calling the result of `configuration.getOutgoing()` in a local named `publications` (or maybe even not plural?) makes the process clearer.